lib/gcloud/dns/zone.rb

```diff
@@ -355,24 +373,25 @@ def record name, type, ttl, data
       #   dns = gcloud.dns
       #   zone = dns.zone "example-zone"
       #   new_record = zone.record "example.com.", "A", 86400, ["1.2.3.4"]
-      #   old_record = zone.record "example.com.", "A", 86400, ["1.2.3.4"]
+      #   old_record = zone.record "example.com.", "A", 18600, ["1.2.3.4"]
       #   zone.update [new_record], [old_record]
       #
-      def update records_to_add = [], records_to_remove = []
-        records_to_add = Array(records_to_add).map(&:to_gapi)
-        records_to_remove = Array(records_to_remove).map(&:to_gapi)
+      def update additions = [], deletions = []
+        additions = Array additions
+        deletions = Array deletions
 
-        ensure_connection!
-        resp = connection.create_change id, records_to_add, records_to_remove
-        if resp.success?
-          Change.from_gapi resp.data, self
-        else
-          fail ApiError.from_response(resp)
+        if block_given?
+          updater = Zone::Transaction.new self
+          yield updater
+          additions += updater.additions
+          deletions += updater.deletions
         end
+
+        create_change additions, deletions
       end
 
       ##
-      # Adds records to the Zone. In order to update existing records, or add
+      # Adds a record to the Zone. In order to update existing records, or add
       # and delete records in the same transaction, use #update.
```

Comment:
Yeah, just clarification in the docs. This will only add a single record object, but the single record object can include multiple data values. It depends on how we count records I guess.